osu.Game.Rulesets.Catch.Tests/TestSceneHyperDashColouring.cs

```diff
@@ -0,0 +1,306 @@
+
+            AddStep("clear hyper-dash", () =>
+            {
+                catcherArea.MovableCatcher.SetHyperDashState(1);
+                catcherArea.MovableCatcher.FinishTransforms();
+            });
+
+            AddAssert("hyper-dash colour cleared from catcher", () => catcherArea.MovableCatcher.Colour == Color4.White);
+        }
+
+        [Test]
+        public void TestHyperDashCatcherEndGlowColour()
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("setup catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    RelativePositionAxes = Axes.None,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, false, false, false);
+            });
+
+            AddStep("start hyper-dashing", () => catcherArea.MovableCatcher.SetHyperDashState(2));
+            AddAssert("end-glow catcher sprite has default hyper-dash colour", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == Catcher.DefaultHyperDashColour));
+        }
+
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestCustomHyperDashCatcherEndGlowColour(bool customHyperDashCatcherColour)
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("setup catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    RelativePositionAxes = Axes.None,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, customHyperDashCatcherColour, true, false);
+            });
+
+            AddStep("start hyper-dashing", () => catcherArea.MovableCatcher.SetHyperDashState(2));
+            AddAssert("end-glow catcher sprite use its hyper-dash colour from skin", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == TestSkin.CustomHyperDashAfterColour));
+        }
+
+        [Test]
+        public void TestCustomHyperDashCatcherEndGlowColourFallback()
+        {
+            CatcherArea catcherArea = null;
+
+            AddStep("setup catcher", () =>
+            {
+                Child = setupSkinHierarchy(catcherArea = new CatcherArea
+                {
+                    RelativePositionAxes = Axes.None,
+                    Anchor = Anchor.Centre,
+                    Origin = Anchor.Centre,
+                    Scale = new Vector2(4f),
+                }, true, false, false);
+            });
+
+            AddStep("start hyper-dashing", () => catcherArea.MovableCatcher.SetHyperDashState(2));
+            AddAssert("end-glow catcher sprite colour falls back to catcher colour from skin", () => catcherArea.OfType<Container<CatcherTrailSprite>>().Any(c => c.Colour == TestSkin.CustomHyperDashColour));
```

Comment:
these assert texts are too long. should be a few words

Reply:
So the assert names used to be like: `end-glow sprite colour default-hyperdash`, but I was change-requested to make it more described as the assert name are mostly for describing the assertion on CI in case it fails, although I agree that it is overly-described and it is my mistake not pointing that out in the request conversation.

Reply:
"colour is correct" is fine. if you really need to explain more that isn't obvious from the test method's name, add an inline comment.